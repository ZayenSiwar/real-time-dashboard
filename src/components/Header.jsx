export default function Header() {
  return (
    <header className="bg-gray-800 p-4 border-b border-gray-700">
      <div className="max-w-7xl mx-auto flex items-center justify-between">
        <h1 className="text-2xl font-semibold">Real-Time Ops Dashboard</h1>
        <div className="flex items-center gap-4 text-sm text-gray-300">
          <div>Bonjour, Admin</div>
          <img className="w-9 h-9 rounded-full" src="https://i.pravatar.cc/40" alt="avatar" />
        </div>
      </div>
    </header>
  );
}
