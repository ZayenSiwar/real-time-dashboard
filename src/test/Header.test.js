import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from '../components/Header';

describe('Header Component', () => {
  it('renders without crashing', () => {
    render(<Header />);
    expect(screen.getByRole('banner')).toBeInTheDocument();
  });

  it('displays the correct dashboard title', () => {
    render(<Header />);
    expect(screen.getByText('Real-Time Ops Dashboard')).toBeInTheDocument();
  });

  it('has the correct heading level', () => {
    render(<Header />);
    const heading = screen.getByRole('heading', { level: 1 });
    expect(heading).toHaveTextContent('Real-Time Ops Dashboard');
  });

  it('displays greeting message', () => {
    render(<Header />);
    expect(screen.getByText('Bonjour, Admin')).toBeInTheDocument();
  });

  it('renders avatar image', () => {
    render(<Header />);
    const avatar = screen.getByAltText('avatar');
    expect(avatar).toBeInTheDocument();
    expect(avatar).toHaveAttribute('src', 'https://i.pravatar.cc/40');
    expect(avatar).toHaveClass('w-9 h-9 rounded-full');
  });

  it('has correct styling classes', () => {
    render(<Header />);
    const header = screen.getByRole('banner');
    expect(header).toHaveClass('bg-gray-800 p-4 border-b border-gray-700');
  });

  it('has flex layout for content', () => {
    render(<Header />);
    const container = screen.getByText('Real-Time Ops Dashboard').parentElement;
    expect(container).toHaveClass('flex items-center justify-between');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
