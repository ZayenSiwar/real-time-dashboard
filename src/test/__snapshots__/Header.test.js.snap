// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="bg-gray-800 p-4 border-b border-gray-700"
  >
    <div
      class="max-w-7xl mx-auto flex items-center justify-between"
    >
      <h1
        class="text-2xl font-semibold"
      >
        Real-Time Ops Dashboard
      </h1>
      <div
        class="flex items-center gap-4 text-sm text-gray-300"
      >
        <div>
          Bonjour, Admin
        </div>
        <img
          alt="avatar"
          class="w-9 h-9 rounded-full"
          src="https://i.pravatar.cc/40"
        />
      </div>
    </div>
  </header>
</DocumentFragment>
`;
